scripts/west_commands/run_common.py

```diff
@@ -201,6 +203,18 @@ def do_run_common(command, user_args, user_runner_args):
         if v is not None:
             setattr(args, a, v)
 
+    # If flashing of a signed file is requested, replace the bin and hex
+    # file paths with their signed equivalents, as generated by `west sign`.
+    # e.g.
+    #     build/zephyr/zephyr.bin -> build/zephyr/zephyr.signed.bin
+    #     build/zephyr/zephyr.hex -> build/zephyr/zephyr.signed.hex
+    if args.signed:
+        def path_convert_to_signed(path):
+            split = path.rsplit('.', 1)
+            return f"{split[0]}.signed.{split[1]}"
+        args.bin_file = path_convert_to_signed(args.bin_file)
+        args.hex_file = path_convert_to_signed(args.hex_file)
```

Comment:
This will not work in all cases.

The default sign command will produce: `zephyr.signed.bin` or hex:
https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/west_commands/sign.py#L247

regardless of the bin file name in the runner:
https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/west_commands/sign.py#L216

but if the file name is different: `CONFIG_KERNEL_HEX_NAME=myapp` then the bin/elf/hex files are named `myapp.bin/hex`, and that will be the name of `args.bin_file` / `args.hex_file`, and you convert that into `myapp.signed.bin` but the default signed file is still named `zephyr.signed.bin`

Reply:
I have updated it to point to build_dir/zephyr.signed.{hex,bin}
If it is desired to change the default output of `west sign`, that should be a separate PR.